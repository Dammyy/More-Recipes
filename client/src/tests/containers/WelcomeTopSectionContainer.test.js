import React from 'react';
import { shallow } from 'enzyme';
import { WelcomeTopSectionContainer, mapStateToProps, mapDispatchToProps } from
  '../../containers/WelcomeTopSectionContainer';


describe('WelcomeTopSectionContainer', () => {
  describe('Snapshot', () => {
    it('component should fully render', () => {
      const component = shallow(<WelcomeTopSectionContainer />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Instance methods', () => {
    describe('logout', () => {
      test('', () => {
        const props = {
          authActions: {
            logout: jest.fn(),
          }
        };
        const component = shallow(<WelcomeTopSectionContainer {...props} />);
        component.instance().logout();
        expect(props.authActions.logout).toHaveBeenCalled();
      });
    });
  });

  describe('container functions', () => {
    describe('mapStateToProps', () => {
      it('returns the expected properties', () => {
        const state = {
          getIn: jest.fn()
        };
        const mstp = mapStateToProps(state);
        expect(mstp).toHaveProperty('firstName');
      });
    });
  });
  describe('mapDispatchToProps', () => {
    it('', () => {
      const dispatch = jest.fn();
      const mdtp = mapDispatchToProps(dispatch);
      expect(mdtp).toHaveProperty('authActions');
    });
  });
});
