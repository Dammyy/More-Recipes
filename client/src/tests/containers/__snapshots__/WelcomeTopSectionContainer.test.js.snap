// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeTopSectionContainer Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeTopSectionContainer
    firstName="stuff"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WelcomeTopSetion
        firstName="stuff"
        logout={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "firstName": "stuff",
        "logout": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WelcomeTopSetion
          firstName="stuff"
          logout={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "firstName": "stuff",
          "logout": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
